backend/src/main/java/com/redhat/cloud/notifications/db/repositories/EndpointRepository.java

```diff
@@ -213,8 +218,8 @@ public boolean updateEndpoint(Endpoint endpoint) {
                 "basicAuthentication = :basicAuthentication, " +
                 "disableSslVerification = :disableSslVerification, secretToken = :secretToken WHERE endpoint.id = :endpointId";
 
-        if (endpoint.getType() == EndpointType.EMAIL_SUBSCRIPTION) {
-            throw new RuntimeException("Unable to update an endpoint of type EMAIL_SUBSCRIPTION");
+        if (endpoint.getType() != null && endpoint.getType().isSystemEndpointType) {
+            throw new RuntimeException("Unable to update an system endpoint of type " + endpoint.getType());
```

Comment:
As commented by @josejulio:
```suggestion
            throw new RuntimeException("Unable to update a system endpoint of type " + endpoint.getType());
```